comms/include/blazingdb/concurrency/BlazingThread.h

```diff
@@ -57,16 +57,37 @@ class BlazingThread {
 	std::shared_ptr<BlazingExceptionHolder> exceptionHolder;
 };
 
-class BlazingMutableThread : public BlazingThread{
+namespace detail {
+template <class F, class Tuple, std::size_t... I>
```

Comment:
This code makes sense to me but it might be a bit confusing to some of the more novice developers. Can we add an explanation that explains to people how apply and apply_impl work

Reply:
to be clear i mean just a few sentences